zoltar-metrics/src/main/java/com/spotify/zoltar/metrics/InstrumentedPredictor.java

```diff
@@ -0,0 +1,135 @@
+/*-
+ * -\-\-
+ * zoltar-metrics
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.zoltar.metrics;
+
+import com.spotify.zoltar.FeatureExtractFns.ExtractFn;
+import com.spotify.zoltar.FeatureExtractor;
+import com.spotify.zoltar.Model;
+import com.spotify.zoltar.PredictFns.AsyncPredictFn;
+import com.spotify.zoltar.PredictFns.PredictFn;
+import com.spotify.zoltar.Predictor;
+
+public interface InstrumentedPredictor<InputT, ValueT> extends Predictor<InputT, ValueT> {
```

Comment:
Should we embed metrics into `Predictor` by default instead of having separate `Predictor`?

Reply:
@rav i thought about that, however i wanted metrics to be decoupled from the `zoltar-core`. I also feel that this solution might not be the best one. Let me think a bit more one this 😄